### PR TITLE
8195 Add SPG to Audit Report

### PR DESCRIPTION
**Pull request by prit-cgi**

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java [hunk 1]
```diff
@@ -26,80 +27,131 @@ public class SecurityUtil {
...
 		userInfo.setSessionState(jwt.getClaim(CLAIM_SESSION_STATE));
 		userInfo.setUsername(jwt.getClaim(CLAIM_USERNAME));
 		userInfo.setUserId(jwt.getClaim(CLAIM_SUB));
-		
+
 		return userInfo;
 	}
-	
+
+	public static String loadSPGBasedOnTask(UserInfo userInfo, TransactionType transactionType) {
```
Comment (weskubo-cgi) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java:69]: This is an absolutely minor point but we don't really used the term task anywhere else. Really we're determining the SPB based on the TransactionType.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java [hunk 2]
```diff
@@ -26,80 +27,131 @@ public class SecurityUtil {
...
+					return role;
+				}
+				break;
+			case NAME_SEARCH:
+				if (permissions.contains("AddPermitHolderWOPHN")) {
+					return role;
+				}
+				break;
+			case CONTRACT_INQUIRY:
+				if (permissions.contains("ContractInquiry") || permissions.contains("GetGroupMembersContractAddress")) {
```
Comment (weskubo-cgi) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java:94]: GetGroupMembersContractAddress is not a valid permission. I know it's in SecurityConfig.java but there is an outstanding bug to fix that. Please use GetContractAddress instead.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java [hunk 3]
```diff
@@ -26,80 +27,131 @@ public class SecurityUtil {
...
+					return role;
+				}
+				break;
+			default:
+				if (permissions != null && permissions.contains(transactionType.getValue())) {
+					return role;
+				}
+			}
+		}
+		return "UNAUTHORIZED";
```
Comment (weskubo-cgi) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java:109]: I don't think "UNAUTHORIZED" is quite correct. If we end up here it's more likely an issue with our code/config that an issue with the user's lack of roles. Can we call it "UNKNOWN" instead? And maybe make it a constant UKNWOWN_ROLE = "UNKNOWN".

File: backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/AuditControllerTest.java [hunk 4]
```diff
@@ -61,7 +61,8 @@ public void testGetOrganization() throws Exception {
 	}
 
 	@Test
-	public void testGetAuditReport_withoutOptionalParam() {;
+	public void testGetAuditReport_withoutOptionalParam() {
+		;
```
Comment (weskubo-cgi) [inline backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/AuditControllerTest.java:65]: This wasn't your change but please remove the extraneous semicolon.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java [hunk 5]
```diff
@@ -26,80 +26,135 @@ public class SecurityUtil {
...
+
+	public static String loadSPGBasedOnTransactionType(UserInfo userInfo, TransactionType transactionType) {
+		Map<String, List<String>> rolePermissions = securityProperties.getRolePermissions();
+		List<String> roles = userInfo.getRoles();
+		if (roles.size() == 1) {
+			return roles.get(0);
+		}
+		for (String role : roles) {
+			List<String> permissions = rolePermissions.get(role.toLowerCase());
+			if(permissions == null) {
```
Comment (weskubo-cgi) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java:77]: Missing space before opening parentheses.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java (hunk 5 above)
Comment (prit-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/security/SecurityUtil.java]: done, should be on the new commit.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**